src/ert/_c_wrappers/enkf/ensemble_config.py

```diff
@@ -38,56 +38,45 @@ def _get_abs_path(file):
     return file
 
 
-def _get_filename(file):
-    if file is not None:
-        file = os.path.basename(file)
-    return file
-
-
-def _option_dict(option_list: List[str], offset: int) -> Dict[str, str]:
-    """Gets the list of options given to a keywords such as GEN_DATA.
-
-    The first step of parsing will separate a line such as
-
-      GEN_DATA NAME INPUT_FORMAT:ASCII RESULT_FILE:file.txt REPORT_STEPS:3
+def _parse_options(option_list: List[str], offset: int) -> Dict[str, str]:
+    """Converts a list of options given to keywords such as GEN_DATA into a dictionary.
 
-    into
+    Args:
+        option_list: The list of options. Each option should be a string
+            formatted as 'KEY:VALUE'.
+        offset: The index in the list from where to start parsing the options.
 
-    >>> opts = ["NAME", "INPUT_FORMAT:ASCII", "RESULT_FILE:file.txt", "REPORT_STEPS:3"]
+    Returns:
+        A dictionary where the keys are the options and the values
+        are the arguments for those options.
 
-    From there, _option_dict can be used to get a dictionary of the options:
-
-    >>> _option_dict(opts, 1)
+    >>> options = ["NAME","INPUT_FORMAT:ASCII","RESULT_FILE:file.txt","REPORT_STEPS:3"]
+    >>> _parse_options(options, 1)
     {'INPUT_FORMAT': 'ASCII', 'RESULT_FILE': 'file.txt', 'REPORT_STEPS': '3'}
 
-    Errors are reported to the log, and erroring fields ignored:
+    """
+    options_dict = {}
 
-    >>> import sys
-    >>> logger.addHandler(logging.StreamHandler(sys.stdout))
-    >>> _option_dict(opts + [":T"], 1)
-    Ignoring argument :T not properly formatted should be of type ARG:VAL
-    {'INPUT_FORMAT': 'ASCII', 'RESULT_FILE': 'file.txt', 'REPORT_STEPS': '3'}
+    for option in option_list[offset:]:
+        try:
+            key, value = option.split(":")
 
-    """
-    option_dict = {}
-    for option_pair in option_list[offset:]:
-        if not isinstance(option_pair, str):
+        except ValueError:
             logger.warning(
```

Comment:
This was added a while ago, before we started using ConfigWarning to signal warnings that are supposed to be shown to the user. Could we use `warnings.warn(msg, category=ConfigWarning)` instead of `logger.warning` while we are at it.

Reply:
Just so that we are on the same page.
The code as it is in `main` today would not raise any warnings if this test is run.
That is, it ignores both "FORWARD_INIT:Frue" and "not_an_opt".
My suggested change would still ignore "FORWARD_INIT:Frue" but would raise on "not_an_opt" because it can't split it.

Reply:
It is still not raise though, it just makes the warning visible to the user.

Reply:
I'm just saying that the way the logic works on `main` ignores things that the current code will not ignore.

Reply:
Would be nice to have a test for the new behavior then. Also, if there is a new error that is expected in case of user error, then we should probably reraise it as `ConfigValidationError` to show it correctly to the user.